pandas/io/common.py

```diff
@@ -324,9 +325,51 @@ def _get_handle(path, mode, encoding=None, compression=None):
         else:
             f = open(path, mode)
 
+    if memory_map and hasattr(f, 'fileno'):
+        try:
+            f = MMapWrapper(f)
+        except:  # fallback: leave file handler as is
```

Comment:
That's inconsistent with what's done on the C engine side, where if the lower-level function calls fail, we just return `NULL` for the source as you can see <a href="https://github.com/pydata/pandas/blob/master/pandas/parser.pyx#L612">here<a>.